crates/db_schema/src/source/site.rs

```diff
@@ -13,56 +14,54 @@ pub struct Site {
   pub sidebar: Option<String>,
   pub published: chrono::NaiveDateTime,
   pub updated: Option<chrono::NaiveDateTime>,
-  pub enable_downvotes: bool,
-  pub open_registration: bool,
-  pub enable_nsfw: bool,
   pub icon: Option<DbUrl>,
   pub banner: Option<DbUrl>,
   pub description: Option<String>,
-  pub community_creation_admin_only: bool,
-  pub require_email_verification: bool,
-  pub require_application: bool,
-  pub application_question: Option<String>,
-  pub private_instance: bool,
   pub actor_id: DbUrl,
   pub last_refreshed_at: chrono::NaiveDateTime,
   pub inbox_url: DbUrl,
   pub private_key: Option<String>,
   pub public_key: String,
-  pub default_theme: String,
-  pub default_post_listing_type: String,
-  pub legal_information: Option<String>,
-  pub application_email_admins: bool,
-  pub hide_modlog_mod_names: bool,
+  pub instance_id: InstanceId,
 }
 
-#[derive(Default)]
+#[derive(Clone, TypedBuilder)]
+#[builder(field_defaults(default))]
 #[cfg_attr(feature = "full", derive(Insertable, AsChangeset))]
 #[cfg_attr(feature = "full", diesel(table_name = site))]
-pub struct SiteForm {
+pub struct SiteInsertForm {
+  #[builder(!default)]
   pub name: String,
-  pub sidebar: Option<Option<String>>,
+  pub sidebar: Option<String>,
   pub updated: Option<chrono::NaiveDateTime>,
-  pub enable_downvotes: Option<bool>,
-  pub open_registration: Option<bool>,
-  pub enable_nsfw: Option<bool>,
+  // when you want to null out a column, you have to send Some(None)), since sending None means you just don't want to update that column.
```

Comment:
Maybe better to move that documentation to the Crud trait, otherwise you would have to copy-paste it everywhere.